### ISSUE TITLE
Adding multiple ocaid's in edit UI only adds one

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#395]
Edit UI allows admin to add multiple ocaid's (archive.org ids) but only sets 1.

## To reproduce
1. Edit a book on local env (e.g. http://localhost:8080/books/OL2058361M/The_wit_wisdom_of_Mark_Twainsadf/edit
2. Add 3 random IA ids
3. Save.

**Actual:** Only the first IA id will persist; all others ignored.
**Expected**: ~Support adding multiple OCAIDs.~ UI permits adding only one IA id, in line with the schema.

~Note it's not a requirement of this issue that the Read buttons query all OCAIDs, but we need a way to unblock imports of records like this.~

~Suggestion: add `identifiers['ia']` that stores all OCAIDs, and leaves `.ocaid` for the "main" ocaid for backwards compat.~

**Comment by guyjeangilles** [comment]
@mekarpeles Is this still an issue? If so, are you willing to be assignee? Note: being assignee doesn't necessarily mean you are the one doing the work for the issue but responsible for collecting information needed to devise a solution. From the wiki:

>At submission and prior to triage, the assigned owner is not necessarily the person who will fix the issue (it is not necessarily even established, at that point, if or when the issue will be fixed at all), but rather they are the person who will do as much or as little as needed to handle the issue (asking questions, soliciting input, establishing and updating the priority, checking if it is a duplicate, etc).
>
>Once an issue is labeled State: Work In Progress, the owner is the individual doing the work, or leading/coordinating the group that is doing the work.

**Comment by cdrini** [comment]
(Confirmed this is still an issue)

**Comment by guyjeangilles** [comment]
Assigning jdlrobson (not tagging by request) per slack discussion since it's UI related

**Comment by jdlrobson** [comment]
> Edit UI allows admin to add multiple ocaid's (archive.org ids) but only sets 1.
What edit id?

@cdrini or @mek can somebody expand the description so it's clear to someone who doesn't know how our project works what we want to fix here and how.

**Comment by cdrini** [comment]
@jdlrobson updated; does that look ok?

**Comment by jdlrobson** [comment]
Thank you @cdrini .perfect!

**Comment by mekarpeles** [comment]
@hornc this seems like it may have a front-end portion and a back-end portion, can we assign you as the Lead for this one?

**Comment by RayBB** [comment]
This can be fixed by adding something as simple as:
```javascript
if (['ocaid'].includes(data.name) && document.getElementsByName("edition--identifiers--0--name").length === 1) {
    return error("#id-errors", "id-value", "$_('ID can only be used once per edition')".replace(/ID/, label));
}
```

However, before that happens, I believe the inline js should be moved from `openlibrary/templates/books/edit/edition.html` per #4474. 

Additionally, if #5093 is merged then it would probably make sense to use that component and do validation in Vue.

**Comment by jdlrobson** [comment]
> However, before that happens, I believe the inline js should be moved from openlibrary/templates/books/edit/edition.html per #4474.

Yep doing that first would be great (and that would include removal of the jsdef). It would also allow us to add a unit test. Is that something you'd be willing to help out with?

**Comment by RayBB** [comment]
@Eds-Dbug are you [still interested](https://github.com/internetarchive/openlibrary/issues/8295#issuecomment-1734558493) in working on this? If so, let me know if you need any help :)

**Comment by hornc** [comment]
`ocaid` is a single string value in current schema:  https://github.com/internetarchive/openlibrary/blob/12d140f9bf8fe54018b1dc292ecdf9d5646843d1/openlibrary/schemata/edition.schema.json#L51C6-L56

**Comment by cdrini** [comment]
The bug here is with the UI not adhering to the schema; it presumably lets you add multiple IA IDs, which it shouldn't let you do. Dropping in priority though since adding IA identifiers is now restricted to accounts with elevated permissions.

**Comment by Eds-Dbug** [comment]
If this issue remains unresolved, no promises yet but I might pick it up.

**Comment by RayBB** [comment]
@Eds-Dbug go for it!

**Comment by Eds-Dbug** [comment]
Hi just started the issue, I am currently looking at the page... would like to confirm this is the correct behaviour. I am trying to add another Internet Archive ID but the selection is greyed out so I assume everything is inorder. Let me know.

<img width="803" height="986" alt="Image" src="https://github.com/user-attachments/assets/f03c8daa-aa6d-4cae-8499-885e102dd725" />

**Comment by Shikhar-395** [comment]
@RayBB I would like to work on this.

**Comment by RayBB** [comment]
Great! You are now assigned
